shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -819,4 +815,13 @@ std::optional<LRESULT> FlutterWindowsEngine::ProcessExternalWindowMessage(
   return std::nullopt;
 }
 
+void FlutterWindowsEngine::OnChannelUpdate(const std::string& name,
+                                           bool listening) {
+  if (name.compare("flutter/platform") == 0) {
```

Comment:
Consider checking that `listening` is `true`. This makes this logic a bit more resilient if the engine ever misbehaved and sent updates `listening` of `false` erroneously.